File: 01. Leetcode/faangdsa/sachin/arrays/07/twoSum.ts
function twoSum(nums: number[], target: number): number[] {
    const seen = new Map<number, number>();

    for (let idx=0; idx<nums.length; idx++){
        let k = target - nums[idx];
        if (seen.has(k)){
            return [seen.get(k)!, idx];
        }
        seen.set(nums[idx], idx);
    }
    return []
};

File: 01. Leetcode/faangdsa/sachin/arrays/07/two_sum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for idx, num in enumerate(nums):
            k = target - num
            if k in seen:
                return [seen[k], idx]
            seen[num] = idx